java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -271,20 +316,29 @@ public static synchronized void initialize(int allocationMode, LogConf logConf,
    * @return true if we were able to setup the default resource, false if there was
    *         a resource already set.
    */
-  public static synchronized native boolean configureDefaultCudfPinnedPoolSize(long size);
+  public static boolean configureDefaultCudfPinnedPoolSize(long size) {
+    writeLock.lock();
+    try {
+      return configureDefaultCudfPinnedPoolSizeImpl(size);
+    } finally {
+      writeLock.unlock();
+    }
+  }
+
+  private static native boolean configureDefaultCudfPinnedPoolSizeImpl(long size);
 
   /**
    * Get the most recently set pool size or -1 if RMM has not been initialized or pooling is
    * not enabled.
    */
-  public static synchronized long getPoolSize() {
+  public static long getPoolSize() {
     return poolSize;
```

Comment:
Is the read lock needed here? Just to protect the poolSize variable from being modified while it is read? This is likely just a nit though.